bpi.syndicate.inc

```diff
@@ -80,35 +176,49 @@ function bpi_syndicate_page_form($form, &$form_state) {
 }
 
 /**
- * Syndication page form validator.
- */
-function bpi_syndicate_page_form_validate($form, &$form_state) {
-  if (empty($form_state['values']['bpi_search_query'])) {
-    form_set_error('bpi_search_query', t('Enter some keywords.'));
-  }
-}
-
-/**
  * Custom submit handler for syndication page form.
  *
- * @see bpi_syndicate_page_form()
- *
  * @param array $form
  *   Form structure.
  * @param array $form_state
  *   Form state values.
  *
+ * @see bpi_syndicate_page_form()
+ *
  * @ingroup forms
  */
-function bpi_syndicate_page_form_submit($form, &$form_state) {
-  $query_parameters = array(
-    'query' => array(
-      BPI_SEARCH_PHRASE_KEY => $form_state['values']['bpi_search_query'],
-    ),
-  );
+function bpi_syndicate_page_form_submit(array $form, array &$form_state) {
+  $bpi_query = array();
+  $sorting = _bpi_get_current_sorting();
+  if ($sorting) {
+    $bpi_query[BPI_QUERY_SORT_NAME] = $sorting;
+  }
+  $values = $form_state['values'];
+
+  if (isset($values['bpi_search_amount']) && intval($values['bpi_search_amount']) > 0) {
+    $bpi_query['amount'] = intval($values['bpi_search_amount']);
+  }
+
+  if (isset($values['bpi_search_query']) && trim($values['bpi_search_query'])) {
+    $bpi_query[BPI_SEARCH_PHRASE_KEY] = trim($values['bpi_search_query']);
+  }
+
+  if (isset($values['bpi_facets'])) {
+    foreach ($values['bpi_facets'] as $key => $value) {
+      $selected_values = is_array($value) ? array_keys(array_filter($value, function($value) {
```

Comment:
This could be changed to array_keys(array_filter($value)) as array_filter as default remove empty values.